api/v3/FinancialTrxn.php

```diff
@@ -39,7 +39,10 @@
  * @return array
  */
 function civicrm_api3_financial_trxn_create($params) {
-  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (empty($params['contribution_id']) && (empty($params['entity_table']) && empty($params['entity_id']))) {
```

Comment:
I think this be 

civicrm_api3_verify_mandatory($params, NULL, array(array('contribution_id', array('entity_table', 'entity_id')));

verify_mandatory can do OR - I Think I got that right for this OR this AND that